Sources/DDGSync/SyncFeatureFlags.swift

```diff
@@ -24,6 +24,7 @@ import Foundation
  */
 public struct SyncFeatureFlags: OptionSet {
     public let rawValue: Int
+    public var unavailableReason: PrivacyConfigurationFeatureDisabledReason?
```

Comment:
I like this approach a lot. I didn't think about it initially because I'm having this mindset that OptionSet is basically an enum so we shouldn't be adding other fields to it, besides the raw value. But since enums have associated values, why not add an extra field to an OptionSet struct if it's needed. It looks like an equivalent of an enum's associated value. 

Good job! 👏

Reply:
I know what you mean lol… initially it didn’t come to me intuitively to do that… and then I thought… why not! we do it for enum too ahahah